### PR TITLE
(Liver) Organ Damage: Begone Toxins

### PR DESCRIPTION
**Pull request by Geevies**
* The liver now slowly filters out toxins once every few seconds.
* If the toxin count is higher than 3, it will take some minor damage whilst filtering, UNLESS there is anti toxin in the body, in which case no organ damage occurs.
* Non-functioning livers will not filter out toxins, and a toxin count higher than 60 will begin damaging other organs, leading to total organ failure and eventually death.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/modules/organs/organ_internal.dm [hunk 1]
```diff
@@ -130,7 +130,25 @@
...
+		if(owner.getToxLoss() > 0 && owner.getToxLoss() <= 3)
+			owner.adjustToxLoss(-0.2) //there isn't a lot of toxin damage, so we're going to be chill and slowly filter it out
+		else if(owner.getToxLoss() > 3)
+			if(is_bruised())
+				//damaged liver works less efficiently
+				owner.adjustToxLoss(-0.5)
+				if(!owner.reagents.has_reagent("anti_toxin")) //no damage to liver if anti-toxin is present
+					src.damage += 0.1 * PROCESS_ACCURACY
+			else if(is_broken())
+				//non-functioning liver doesn't fix toxins
```
Comment (NonQueueingMatt) [inline code/modules/organs/organ_internal.dm]: empty else statement?

File: code/modules/organs/organ_internal.dm [hunk 2]
```diff
@@ -130,7 +130,25 @@
...
 	if(owner.life_tick % PROCESS_ACCURACY == 0)
 
-		//High toxins levels are dangerous
+		//A liver's duty is to get rid of toxins
+		if(owner.getToxLoss() > 0 && owner.getToxLoss() <= 3)
+			owner.adjustToxLoss(-0.2) //there isn't a lot of toxin damage, so we're going to be chill and slowly filter it out
+		else if(owner.getToxLoss() > 3)
+			if(is_bruised())
+				//damaged liver works less efficiently
+				owner.adjustToxLoss(-0.5)
```
Comment (NonQueueingMatt) [inline code/modules/organs/organ_internal.dm]: removing 0.5 instead of 0.2 seems more efficient to me?

Comment (Geevies) [reply code/modules/organs/organ_internal.dm]: A healthy liver removes 1 tox per one of these if there's more than 3 toxins in the body. A bruised liver removes 0.5 per one of these.

File: code/modules/organs/organ_internal.dm (hunk 1 above)
Comment (Geevies) [reply code/modules/organs/organ_internal.dm]: Would it be better to have it deal toxin damage, instead of doing nothing?

Comment (NonQueueingMatt) [reply code/modules/organs/organ_internal.dm]: i would say so tbh

Comment (Geevies) [reply code/modules/organs/organ_internal.dm]: commit die

**Comment by sheeplets** [comment]
Three toxin damage sounds like a really low ceiling for organ damage. Like, dumping a drink on yourself at that rate would just make your liver die. I can only imagine how much Peridaxon is going to get used per round.

**Comment by Geevies** [comment]
> 
> 
> Three toxin damage sounds like a really low ceiling for organ damage. Like, dumping a drink on yourself would be enough to just make your liver die. Not to mention how deadly scrubbers blasts will be, or how quickly K'ois bars and Phoron Punch will off any unsuspecting baldies. Medical is going to go through an insane amount of Peridaxon.

I thought it was super low too, but in my testing, most sources of toxin damage gives less than three toxins damage, which quickly get nullified too quickly. It takes around 30 units of organ brute damage for the liver to really start showing major issues. 
I've tested it on a few things, including attacking spiders with a stun baton, and setting up the supermatter with no radiation protection. It works rather well, and you don't end up with a TON of brute damage.

I respond quickly to feedback, so if some edge case pops up where this really makes the QoL worse, I'll quickly patch it. <3

**Comment by Bygonehero** [comment]
My concern is that this will nerf Vaurca into the ground with their toxin vulnerability

**Comment by Conspiir** [comment]
My understanding is that the anti_toxin doesn't take into account dyn, which isn't actually dylovene but has the same (weaker) effect. 

Is this a choice or just overlooked?

**Comment by MarinaGryphon** [comment]
> My concern is that this will nerf Vaurca into the ground with their toxin vulnerability

Then request a change.

**Comment by Geevies** [comment]
> 
> 
> > My concern is that this will nerf Vaurca into the ground with their toxin vulnerability
> 
> Then request a change.

No, no. It's fine, we actually had a chat in DMs that cleared some stuff up. Vaurca balance was something I hadn't considered before, but I still think this wouldn't hurt them too much. Naturally, the results in-game may vary, but in my testing it didn't cause any issues.
IF it does, I can fix it extremely quickly.

**Pull request merged**